docs/fsharp-cheatsheet.md

```diff
@@ -540,17 +538,156 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns*:
 
-    let (|DivisibleBy|_|) by n = 
+    let (|DivisibleBy|_|) by n =
         if n % by = 0 then Some DivisibleBy else None
-    
-    let fizzBuzz = function 
-        | DivisibleBy 3 & DivisibleBy 5 -> "FizzBuzz" 
-        | DivisibleBy 3 -> "Fizz" 
-        | DivisibleBy 5 -> "Buzz" 
+
+    let fizzBuzz = function
+        | DivisibleBy 3 & DivisibleBy 5 -> "FizzBuzz"
+        | DivisibleBy 3 -> "Fizz"
+        | DivisibleBy 5 -> "Buzz"
         | i -> string i
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<a name="CodeOrganization"></a>Code Organization
+---------
+
+<a name="CodeOrganization_Modules"></a>
+### Modules
+Modules are key building blocks for grouping related code; they can contain `types`, `let` bindings, or (nested) sub `module`s.
+Identifiers within modules can be referenced using dot notation, or you can bring them into scope via the `open` keyword. Illustrative-only example:
+
+    module Game =
+        let mutable basePoints = 1
+        type Player = { id: int; score: int }
+        let playerScored player = { player with score = player.score + basePoints }
+
+    let player: Game.Player = { id = 1; score = 0 }
+    let player' = Game.playerScored player  // score = 1
+
+    open Game
+    basePoints <- 2
+    let player'' = playerScored player'  // player''.score = 3
+
+If you have only one module in a file, the `module` name can be declared at the top of the file, with all further declarations
+being module elements (and non indentation required)
+
+    module Functions  // notice there is no '=' when at the top of a file
+
+    let addToFive num = 5 + num
+    let subtractFive num = num - 5
+
+### Namespaces
+Namespaces are simply dotted names that prefix other program elements to allow for further hierarchical organization.
```

Comment:
i think the program elements are types and modules, what else?
also maybe prefix -> scope?

Reply:
Good point, I'll specify as there are only two.

Explaining this simply is proving tricky. Will try again.